File: tests/fixtures/sample.go
// Package sample provides test fixtures for tree-sitter Go parsing.
// It demonstrates functions, methods, structs, and interfaces.
package sample

import (
	"fmt"
	"net/http"
)

// UserService manages user operations.
// It provides CRUD functionality.
type UserService struct {
	Name  string
	Email string
}

// GetName returns the user's display name.
func (s *UserService) GetName() string {
	return s.Name
}

// unexportedMethod is private to the package.
func (s *UserService) unexportedMethod() {
	// internal logic
}

// Handler is the HTTP handler interface.
type Handler interface {
	ServeHTTP(w http.ResponseWriter, r *http.Request)
}

// Greet returns a greeting message for the given name.
func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// helper is not exported.
func helper() int {
	return 42
}

File: tests/fixtures/sample.ts
/**
 * Sample TypeScript module for testing tree-sitter parsing.
 * Demonstrates functions, classes, interfaces, and imports.
 */

import { EventEmitter } from 'events';
import type { Config } from './config';

/** Greeting helper that returns a formatted string. */
export function greet(name: string): string {
    return `Hello, ${name}!`;
}

function _privateHelper(): void {
    // not exported
}

/**
 * Represents a user in the system.
 * Tracks name and email.
 */
export class UserService {
    private name: string;

    constructor(name: string) {
        this.name = name;
    }

    getName(): string {
        return this.name;
    }

    updateEmail(email: string): void {
        // update logic
    }
}

/** Configuration options for the app. */
export interface AppConfig {
    port: number;
    host: string;
    debug: boolean;
}

function calculateScore(items: number[]): number {
    return items.reduce((a, b) => a + b, 0);
}
